### PR TITLE
Fix crash when locale is invalid

### PR DESCRIPTION
**Pull request by yufeih**
https://github.com/dotnet/docfx/pull/3441

### CODE REVIEW

File: src/docfx/config/Config.cs [hunk 1]
```diff
@@ -26,8 +26,8 @@ internal sealed class Config
 
         /// <summary>
         /// Gets the default locale of this docset.
+        /// TODO: lower case user inputs.
         /// </summary>
-        [Locale]
```
Comment (OsmondJiang) [inline src/docfx/config/Config.cs]: >Locale [](start = 9, length = 6)

Some pipeline can be run without docset, how can we prevent these errors

File: src/Microsoft.Docs.Template/Models/Attributes.cs [hunk 2]
```diff
@@ -36,21 +34,4 @@ public class HtmlAttribute : DataTypeAttribute { }
...
-        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
-        {
-            try
-            {
-                var culture = new CultureInfo(value.ToString());
-                return null;
-            }
-            catch (CultureNotFoundException)
-            {
-                return new ValidationResult($"Locale '{value.ToString()}' is not supported.", null);
```
Comment (OsmondJiang) [inline src/Microsoft.Docs.Template/Models/Attributes.cs]: >ValidationResult [](start = 27, length = 16)

why not just throw DocfxException here?

Comment (live1206) [reply src/Microsoft.Docs.Template/Models/Attributes.cs]: +1

---
In reply to: [219429736](https://github.com/dotnet/docfx/pull/3444#discussion_r219429736) [](ancestors = 219429736)

Comment (yufeih) [reply src/Microsoft.Docs.Template/Models/Attributes.cs]: Data annotation is primarily used for SDP as an extension mechanism where all the logic is self-contained inside the implementation of a particular attribute.

**Comment by yufeih** [comment]
According to https://github.com/dotnet/corefx/issues/6374, `CultureInfo` depend on OS, and there is no predefined list of **all valid locales**

**Pull request merged**